### PR TITLE
[HttpFoundation] Fix setting session-related ini settings

### LINKED ISSUE #27011: Session ini_set bug ?
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | yes/no
| Symfony version  | 4.x

Hi,

After looking at the code, I think there is two small bugs in the session class
https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php#L356

There is the possibilite to apply a "url_rewriter.tags" parameter value, but this parameter doesn't start with "session.xxxx" so i think this will not work.

And there is a "upload_progress.min-freq" but looking at the doc it should be an underscore beetween min and fred => upload_progress.min_freq if i look at the doc : http://php.net/manual/fr/session.configuration.php#ini.session.upload-progress.min-freq

Am i missing something ? or are these 2 small potential bugs ?

### PR DESCRIPTION
**Pull request by e-moe**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes 
| Fixed tickets | #27011 
| License       | MIT
| Doc PR        | n/a

Added missed option `cache_expire`
Fixed typo in `upload_progress.min_freq`
Fixed ini_set name prefix of `url_rewriter.tags`

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 1]
```diff
@@ -340,21 +340,23 @@ public function setOptions(array $options)
         }
 
         $validOptions = array_flip(array(
-            'cache_limiter', 'cookie_domain', 'cookie_httponly',
+            'cache_limiter', 'cache_expire', 'cookie_domain', 'cookie_httponly',
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: can you move it after cookie_domain? would make it alpha order

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 2]
```diff
@@ -340,21 +340,23 @@ public function setOptions(array $options)
...
-            'upload_progress.freq', 'upload_progress.min-freq', 'url_rewriter.tags',
+            'upload_progress.freq', 'upload_progress.min_freq', 'url_rewriter.tags',
             'sid_length', 'sid_bits_per_character', 'trans_sid_hosts', 'trans_sid_tags',
         ));
 
         foreach ($options as $key => $value) {
             if (isset($validOptions[$key])) {
-                ini_set('session.'.$key, $value);
+                // All options starts with 'session.' except 'url_rewriter.tags'
+                $name = 'url_rewriter.tags' != $key ? 'session.'.$key : $key;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: should use `!==`

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 3]
```diff
@@ -340,21 +340,23 @@ public function setOptions(array $options)
...
+                ini_set($name, $value);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: let's inline?
`ini_set('url_rewriter.tags' !== $key ? 'session.'.$key : $key, $value);`

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 4]
```diff
@@ -340,21 +340,23 @@ public function setOptions(array $options)
...
             'upload_progress.cleanup', 'upload_progress.prefix', 'upload_progress.name',
-            'upload_progress.freq', 'upload_progress.min-freq', 'url_rewriter.tags',
+            'upload_progress.freq', 'upload_progress.min_freq', 'url_rewriter.tags',
             'sid_length', 'sid_bits_per_character', 'trans_sid_hosts', 'trans_sid_tags',
         ));
 
         foreach ($options as $key => $value) {
             if (isset($validOptions[$key])) {
-                ini_set('session.'.$key, $value);
+                // All options starts with 'session.' except 'url_rewriter.tags'
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: I think we should remove this comment as it just duplicates what the code tells quite evidently

File: src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php [hunk 5]
```diff
@@ -183,6 +183,19 @@ public function testCookieOptions()
...
+    {
+        $options = array(
+            'url_rewriter.tags' => 'a=href',
+            'cache_expire' => 200,
+        );
+
+        $this->getStorage($options);
+
+        $this->assertEquals('a=href', ini_get('url_rewriter.tags'));
+        $this->assertEquals(200, ini_get('session.cache_expire'));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php]: please use assertSame in both cases

**Review by nicolas-grekas — APPROVED** [review]
thanks for the PR, here are some minor comments

File: src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php [hunk 6]
```diff
@@ -183,6 +183,19 @@ public function testCookieOptions()
...
+    public function testSessionOptions()
+    {
+        $options = array(
+            'url_rewriter.tags' => 'a=href',
+            'cache_expire' => 200,
+        );
+
+        $this->getStorage($options);
+
+        $this->assertEquals('a=href', ini_get('url_rewriter.tags'));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Session/Storage/NativeSessionStorageTest.php]: This fails on HHVM. I suggest to skip this test on HHVM.:
```php
if (defined('HHVM_VERSION')) {
    $this->markTestSkipped('HHVM is not handled in this test case.');
}

**Review by nicolas-grekas — APPROVED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Good catch, thanks @e-moe.

**Pull request merged**